### ISSUE TITLE
All function names in output of v0.2.0-alpha.1 are 'undefined'

### ISSUE THREAD

**Issue by barakman** [OpenZeppelin/solidity-docgen#32]
This is a bug in file `extract.js` function `getOwnFunctions`.

The `astNode` input seems to contain only a single value, hence in:
```
    const {
      name,
      kind
    } = astNode;
```
The value of `kind` is always `undefined`.

Subsequently, in:
```
    const isRegularFunction = kind === 'function';
    const actualName = isRegularFunction ? name : kind;
```
The value of `actualName ` is always `undefined`.

Following [my previous issue](https://github.com/OpenZeppelin/solidity-docgen/issues/30), one can "hot-fix" this by adding the following lines in file `fix-modules.js`:
```
fix("./node_modules/solidity-docgen/dist/gather/solidity/extract.js", [
    {prev: "? name : kind", next: "? name : name"}]
);
```
The names of the constructor and the fallback function still need to be dealt with (instead of appearing as `undefined`, they now appear empty).

Thanks

**Comment by frangio** [comment]
Hm, I have a suspicion that this may be due to different output of solc 0.4. I'll look into this and report back.

**Comment by barakman** [comment]
@frangio:

Thanks.

So it is indeed worth mentioning that prior to this fix, I changed `_solc.default.compile` to `_solc.default.compileStandard` in order to support Solc v0.4.x.

This change is described in [a previous issue that I have posted](https://github.com/OpenZeppelin/solidity-docgen/issues/30) (which happens to be the same one that I have linked above, though I linked it there for a different reason).

I hope that `compileStandard` works for all compiler versions (or by the least for 0.4.x and above, which I believe to be the most common versions in use at present), which will allow you to keep this proposed change and support v0.4.x.

Thanks
